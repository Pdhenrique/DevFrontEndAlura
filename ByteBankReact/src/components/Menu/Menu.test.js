import { render, screen } from '@testing-library/react';
import Menu from './index';

test('should render a home page link', () => {
  render(<Menu />);
  const homePageLink = screen.getByText('Home');

  expect(homePageLink).toBeInTheDocument();
});

test('should render a list of links', () => {
  render(<Menu />);
  const linksList = screen.getAllByRole('link');

  expect(linksList).toHaveLength(4);
});

test("Shouldn't render link to Extract", () => {
  render(<Menu />);
  const LinksExtract = screen.queryByText('Extract');

  expect(LinksExtract).not.toBeInTheDocument();
});

test('should render a list of links with the class link', () => {
  render(<Menu />);
  const links = screen.getAllByRole('link');

  links.forEach((link) => expect(link).toHaveClass('link'));
  expect(links).toMatchSnapshot();
});
