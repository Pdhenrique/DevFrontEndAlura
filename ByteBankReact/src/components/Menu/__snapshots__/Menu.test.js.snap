// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a list of links with the class link 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Home
  </a>,
  <a
    class="link"
    href="/"
  >
    Transaction
  </a>,
  <a
    class="link"
    href="/"
  >
    Investment
  </a>,
  <a
    class="link"
    href="/"
  >
    Other works
  </a>,
]
`;
